qiita_db/util.py

```diff
@@ -1219,3 +1219,159 @@ def supported_filepath_types(artifact_type):
+                        study_id = qiita.study.study_id) AS artifact_biom_ts,
+                (SELECT array_agg(DISTINCT visibility) FROM qiita.artifact
+                    LEFT JOIN qiita.visibility USING (visibility_id)
+                    WHERE study_id=qiita.study.study_id)
+                    AS artifacts_visibility,
+                (SELECT array_agg(publication_doi ORDER BY publication_doi)
+                    FROM qiita.study_publication
+                    WHERE study_id=qiita.study.study_id) AS publication_doi,
+                (SELECT array_agg(name ORDER BY email) FROM qiita.study_users
+                    LEFT JOIN qiita.qiita_user USING (email)
+                    WHERE study_id=qiita.study.study_id) AS shared_with_name,
+                (SELECT array_agg(email ORDER BY email) FROM qiita.study_users
+                    LEFT JOIN qiita.qiita_user USING (email)
+                    WHERE study_id=qiita.study.study_id) AS shared_with_email
+                FROM qiita.study
+                LEFT JOIN qiita.study_person ON (
+                    study_person_id=principal_investigator_id)
+                WHERE study_id IN %s"""
+        qdb.sql_connection.TRN.add(sql, [tuple(study_ids)])
+        infolist = []
+        refs = {}
+        for info in qdb.sql_connection.TRN.execute_fetchindex():
+            info = dict(info)
+
+            # publication info
+            if info['publication_doi'] is not None:
+                info['pmid'] = get_pubmed_ids_from_dois(
+                    info['publication_doi']).values()
+            else:
+                info['publication_doi'] = []
+                info['pmid'] = []
+
+            # visibility
+            info["status"] = infer_status([info['artifacts_visibility']])
+            del info['artifacts_visibility']
+
+            # pi info
+            info["pi"] = (info['pi_email'], info['pi_name'])
+            del info["pi_email"]
+            del info["pi_name"]
+
+            # shared with
+            info['shared'] = []
+            if info['shared_with_name']:
+                for name, email in zip(info['shared_with_name'],
+                                       info['shared_with_email']):
+                    if not name:
+                        name = email
+                    info['shared'].append((email, name))
+            del info["shared_with_name"]
+            del info["shared_with_email"]
+
+            info['proc_data_info'] = []
+            if build_samples and info['artifact_biom_ids']:
+                to_loop = zip(
+                    info['artifact_biom_ids'], info['artifact_biom_dts'],
+                    info['artifact_biom_ts'], info['artifact_biom_params'])
+                for artifact_id, dt, ts, params in to_loop:
+                    proc_info = {'processed_date': str(ts)}
+                    proc_info['pid'] = artifact_id
+                    proc_info['data_type'] = dt
+                    del params['input_data']
```

Comment:
1. I will add a way to automatically remove everything that is artifact in the parameter_type from qiita.command_parameter
2. If there is no tree it will return a None, which is fine.
3-4. Other processing pipelines - direct BIOM uploads: I guess that depends on how those are implemented and we will need to solve once they are added.